extensions/quickfile/src/utils/hooks.ts

```diff
@@ -0,0 +1,46 @@
+import { showToast, Toast, LocalStorage } from "@raycast/api";
+import { useEffect, useState } from "react";
+
+export const useAsyncData = <T = any, I extends string | string[] | number[] = string>(
+  _key: I,
+  fetcher: (key: I) => Promise<T>
+) => {
+  const [data, setData] = useState<T>();
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<any>();
+
+  const key: string = Array.isArray(_key) ? _key.join(",") : _key;
+  useEffect(() => {
+    if (!key) return;
+
+    Promise.race([
```

Comment:
would it make sense to use [Promise.any](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/any) here instead? In case the first promise (`LocalStorage.getItem`) rejects due to e.g. `JSON.parse` throwing an exception, we'd probably want to wait for the API call to resolve, only _then_ resolving the parent promise.

Reply:
Agreed 👍 Nice catch.